src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Sql/Sprocs/SwitchPartitionsOut.sql

```diff
@@ -0,0 +1,136 @@
+DECLARE @CheckConstraints TABLE (CheckName varchar(200), CheckDefinition varchar(200))
+
+BEGIN TRY
+  EXECUTE dbo.LogEvent @Process=@SP,@Mode=@Mode,@Status='Start'
+
+  IF @Tbl IS NULL RAISERROR('@Tbl IS NULL', 18, 127)
+  IF @RebuildClustered IS NULL RAISERROR('@RebuildClustered IS NULL', 18, 127)
+
+  INSERT INTO @Indexes SELECT index_id, name, is_disabled FROM sys.indexes WHERE object_id = object_id(@Tbl) AND (is_disabled = 0 OR @RebuildClustered = 1)
+  EXECUTE dbo.LogEvent @Process=@SP,@Mode=@Mode,@Status='Info',@Target='@Indexes',@Action='Insert',@Rows=@@rowcount
+
+  INSERT INTO @ResourceTypes 
+    SELECT ResourceTypeId = partition_number - 1
+          ,partition_number_roundtrip = $partition.PartitionFunction_ResourceTypeId(partition_number - 1)
+          ,partition_number
+          ,row_count
+      FROM sys.dm_db_partition_stats 
+      WHERE object_id = object_id(@Tbl) 
+        AND index_id = 1
+        AND row_count > 0
+  EXECUTE dbo.LogEvent @Process=@SP,@Mode=@Mode,@Status='Info',@Target='@ResourceTypes',@Action='Insert',@Rows=@@rowcount,@Text='For partition switch'
+
+  -- Sanity check
+  IF EXISTS (SELECT * FROM @ResourceTypes WHERE partition_number_roundtrip <> partition_number) RAISERROR('Partition sanity check failed', 18, 127)
+
+  WHILE EXISTS (SELECT * FROM @ResourceTypes)
+  BEGIN
+    SELECT TOP 1 @ResourceTypeId = ResourceTypeId, @Rows = row_count FROM @ResourceTypes ORDER BY ResourceTypeId
+    SET @TblInt = @Tbl+'_'+convert(varchar,@ResourceTypeId)
+    SET @Txt = 'Starting @ResourceTypeId='+convert(varchar,@ResourceTypeId)+' row_count='+convert(varchar,@Rows)
+    EXECUTE dbo.LogEvent @Process=@SP,@Mode=@Mode,@Status='Info',@Text=@Txt
+
+    IF NOT EXISTS (SELECT * FROM sysindexes WHERE id = object_id(@TblInt) AND rows > 0)
+    BEGIN
+      IF object_id(@TblInt) IS NOT NULL
+      BEGIN
+        EXECUTE('DROP TABLE dbo.'+@TblInt)
+        EXECUTE dbo.LogEvent @Process=@SP,@Mode=@Mode,@Status='Info',@Target=@TblInt,@Action='Drop'
+      END
+
+      EXECUTE('SELECT * INTO dbo.'+@TblInt+' FROM dbo.'+@Tbl+' WHERE 1 = 2')
+      EXECUTE dbo.LogEvent @Process=@SP,@Mode=@Mode,@Status='Info',@Target=@TblInt,@Action='Select Into',@Rows=@@rowcount
+      DELETE FROM @CheckConstraints
+      
+      INSERT INTO @CheckConstraints SELECT name, definition FROM sys.check_constraints WHERE parent_object_id = object_id(@Tbl) 
+      WHILE EXISTS (SELECT * FROM @CheckConstraints)
+      BEGIN
+        SELECT TOP 1 @checkName=CheckName, @definition=CheckDefinition from @CheckConstraints
+        SET @Txt = 'ALTER TABLE '+@TblInt+' ADD CHECK '+@definition
+        EXECUTE(@Txt)
+        EXECUTE dbo.LogEvent @Process=@SP,@Mode=@Mode,@Status='Info',@Target=@TblInt,@Action='ALTER',@Text=@Txt
+
+        DELETE FROM @CheckConstraints WHERE CheckName=@checkName
+      END
+      DELETE FROM @Names
+
+      INSERT INTO @Names SELECT name FROM sys.columns WHERE object_id = object_id(@Tbl) AND is_sparse = 1
```

Comment:
Preferred line separation is below

Preferred line separation is shown below

END...

DELETE...
INSERT...

Reply:
Ok, I will modify such lines.